pallets/did/src/did_details.rs

```diff
@@ -70,21 +72,47 @@ impl DidVerificationKey {
 			_ => Err(errors::SignatureError::InvalidFormat),
 		}
 	}
+
+	pub fn into_account(&self) -> AccountId
+	where
+		AccountId: From<[u8; 32]> + AsRef<[u8; 32]>,
+	{
+		let bytes = match self {
+			DidVerificationKey::Ed25519(pub_key) => pub_key.0,
+			DidVerificationKey::Sr25519(pub_key) => pub_key.0,
+			// Hash the Ecdsa key the same way it's done in substrate (the ecdsa key is 33 bytes, one byte too long)
+			DidVerificationKey::Ecdsa(pub_key) => sp_io::hashing::blake2_256(pub_key.as_ref()),
+			DidVerificationKey::Account(acc_id) => *acc_id.as_ref(),
+		};
+
+		bytes.into()
+	}
+}
+
+impl<AccountId: AsRef<[u8]>> AsRef<[u8]> for DidVerificationKey<AccountId> {
```

Comment:
Why do we need this implementation here?

Reply:
Yes that was necessary for some old code but not needed anymore. e9ddbd4d5cfbdf22f1fd89ffdeccabfa50088bf4